src/highdicom/legacy/sop.py

```diff
@@ -417,66 +545,1911 @@ def __init__(
             sop_instance_uid=sop_instance_uid,
             sop_class_uid=sop_class_uid,
             instance_number=instance_number,
-            manufacturer=ref_ds.Manufacturer,
-            modality=ref_ds.Modality,
-            transfer_syntax_uid=None,  # FIXME: frame encoding
-            patient_id=ref_ds.PatientID,
-            patient_name=ref_ds.PatientName,
-            patient_birth_date=ref_ds.PatientBirthDate,
-            patient_sex=ref_ds.PatientSex,
-            accession_number=ref_ds.AccessionNumber,
-            study_id=ref_ds.StudyID,
-            study_date=ref_ds.StudyDate,
-            study_time=ref_ds.StudyTime,
-            referring_physician_name=ref_ds.ReferringPhysicianName,
-            **kwargs
+            manufacturer="" if 'Manufacturer' not in ref_ds
+            else ref_ds.Manufacturer,
+            modality="" if 'Modality' not in ref_ds
+            else ref_ds.Modality,
+            patient_id=None if 'PatientID' not in ref_ds
+            else ref_ds.PatientID,
+            patient_name=None if 'PatientName' not in ref_ds
+            else ref_ds.PatientName,
+            patient_birth_date=None if 'PatientBirthDate' not in ref_ds
+            else ref_ds.PatientBirthDate,
+            patient_sex=None if 'PatientSex' not in ref_ds
+            else ref_ds.PatientSex,
+            accession_number=None if 'AccessionNumber' not in ref_ds
+            else ref_ds.AccessionNumber,
+            study_id=None if 'StudyID' not in ref_ds
+            else ref_ds.StudyID,
+            study_date=None if 'StudyDate' not in ref_ds
+            else ref_ds.StudyDate,
+            study_time=None if 'StudyTime' not in ref_ds
+            else ref_ds.StudyTime,
+            referring_physician_name=None if 'ReferringPhysicianName' not in
+            ref_ds else ref_ds.ReferringPhysicianName,
         )
-        _convert_legacy_to_enhanced(legacy_datasets, self)
-        self.PresentationLUTShape = 'IDENTITY'
+        self._legacy_datasets = legacy_datasets
+        self._perframe_functional_group = ـPerframeFunctionalGroup(
+            len(legacy_datasets))
+        tg = tag_for_keyword('PerFrameFunctionalGroupsSequence')
+        self[tg] = DataElement(tg, 'SQ', self._perframe_functional_group)
+        self._shared_functional_group = ـSharedFunctionalGroup()
+        tg = tag_for_keyword('SharedFunctionalGroupsSequence')
+        self[tg] = DataElement(tg, 'SQ', self._shared_functional_group)
+        self.distinguishing_attributes_tags = self._get_tag_used_dictionary(
+            frame_set.distinguishing_attributes_tags)
+        self.excluded_from_perframe_tags = self._get_tag_used_dictionary(
+            frame_set.excluded_from_perframe_tags)
+        self._perframe_tags = self._get_tag_used_dictionary(
+            frame_set.perframe_tags)
+        self._shared_tags = self._get_tag_used_dictionary(
+            frame_set.shared_tags)
+        self.ExcludedFromFunctionalGroupsTags = {
```

Comment:
Is this a DICOM attribute? If not, please use Python naming convention and make "private".